index.js

```diff
@@ -24,106 +25,68 @@ app.get('/info', (req, res) => {
 
 // route for displaying all products
 app.get('/products/all', (req, res) => {
-
-  const jsonOutputObject = {
-    products: []
-  };
-
-  for (let product of productData.products) {
-
-    // Deep copying to prevent changes in original JSON object in next step
-    const productToBeInserted = JSON.parse(JSON.stringify(product));
-
-    // Adding category name to product
-    for (let category of categoryData.categories) {
-      if (product.categoryId === category.id) {
-        productToBeInserted.categoryName = category.categoryName;
-        break;
-      }
-    }
-
-    jsonOutputObject.products.push(productToBeInserted);
-  }
-
-  res.send(jsonOutputObject);
+  res.send({ products: dataService.getCombinedProductMap });
 });
 
 // Getting product info from its ID
 app.get('/product/:id', (req, res) => {
 
   const userInput = req.params.id;
   let productId;
-  let jsonOutputObject = null;
+  let productToBeReturned;
 
   // Generating valid ID and returning error message when ID is invalid
   if (userInput < 1) {
     res.send('Invalid product ID');
     return;
   } else {
-    productId = `P${userInput.padStart(3, '0')}`
+    productId = `P${userInput.padStart(3, '0')}`;
   }
 
-  for (let product of productData.products) {
+  for (product of dataService.getCombinedProductMap) {
     if (product.id === productId) {
-      jsonOutputObject = product;
+      productToBeReturned = product;
       break;
     }
   }
 
-  // Checking for null at the end would cause NullPointerException when trying to get the categoryName
-  if (jsonOutputObject == null) {
+  if (productToBeReturned == null) {
     res.send('Product not found!');
-    return;
-  }
-
-  for (let category of categoryData.categories) {
-    if (jsonOutputObject.categoryId === category.id) {
-      jsonOutputObject.categoryName = category.categoryName;
-    }
+  } else {
+    res.send(productToBeReturned);
   }
-
-  res.send(jsonOutputObject);
 });
 
 // Getting all the products under a category based on category's ID
 app.get('/category/:ctyId', (req, res) => {
 
   const userInput = req.params.ctyId;
   let categoryId;
-  let categoryNameVal;
 
   // Generating valid ID and returning error message when ID is invalid
   if (userInput < 1) {
-    res.send('Invalid category ID')
+    res.send('Invalid category ID');
     return;
   } else {
-    categoryId = `cty${userInput.padStart(2, '0')}`
-  }
-
-  // Getting category name
-  for (let category of categoryData.categories) {
-    if (category.id === categoryId) {
-      categoryNameVal = category.categoryName;
-      break;
-    }
+    categoryId = `cty${userInput.padStart(2, '0')}`;
   }
 
-  const jsonOutputObject = {
-    category: categoryId,
-    categoryName: categoryNameVal,
-    products: []
+  const jsonObjectToBeReturned = {
+    ctyId: categoryId,
+    categoryName: dataService.getCategories[categoryId],
+    products: [],
```

Comment:
Should return a Map in the shape of an object. Initially it should be
```suggestion
    products: {},
```

Reply:
I thaught it would be easier accessing the array, anyway done!